### PR TITLE
Use PersistKeySet to ensure private key is persisted for client auth.

### PR DESCRIPTION
**Pull request by NCarlsonMSFT**

### CODE REVIEW

**Comment by jmeixnerOfStarbucks** [comment]
I can confirm on behalf of Starbucks that this solves our issue. We where able to get 8 completely successful deploys each with 4 Service Fabric applications in it. Thanks @NCarlsonMSFT for your help in finding this issue.

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 [hunk 1]
```diff
@@ -9,271 +9,286 @@ Import-Module $PSScriptRoot\ps_modules\TelemetryHelper
...
-            Start-Sleep -Seconds 3
-            $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
         }
-        Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $newApplication.Status)
-
-        if ($newApplication.Status -ne 'Created' -and $newApplication.Status -ne 'Ready')
+    }
+    finally
+    {
+        if ($null -ne $certificate)
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: There already is a finally block. We can use that. You won't need to add a new try.

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 [hunk 2]
```diff
@@ -9,271 +9,286 @@ Import-Module $PSScriptRoot\ps_modules\TelemetryHelper
...
         {
-            Write-Error (Get-VstsLocString -Key DeployFailed -ArgumentList @($newApplication.Status.ToString(), $newApplication.StatusDetails))
+            $thumbprint = $certificate.Thumbprint
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: This repeated code can be moved to helper module

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 [hunk 3]
```diff
@@ -9,271 +9,286 @@ Import-Module $PSScriptRoot\ps_modules\TelemetryHelper
...
+            $certificate | Get-Member
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: is this line required?

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 [hunk 4]
```diff
@@ -9,271 +9,286 @@ Import-Module $PSScriptRoot\ps_modules\TelemetryHelper
...
+            if (Test-Path "Cert:\CurrentUser\My\$thumbprint")
+            {
+                Remove-Item "Cert:\CurrentUser\My\$thumbprint" -Force
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: Any error in removing cert should be suppressed

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 (hunk 3 above)
Comment (NCarlsonMSFT) [reply Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: Nope, that was for debugging, thanks.

File: Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1 (hunk 4 above)
Comment (NCarlsonMSFT) [reply Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1]: I was debating outright suppressing them, which is why I used two finally blocks (didn't want any error to remove telemetry) will add suppression and remove extra finally.

File: Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1 [hunk 5]
```diff
@@ -185,6 +208,7 @@ function Connect-ServiceFabricClusterFromServiceEndpoint
         try
         {
             [void](Connect-ServiceFabricClusterAction -ClusterConnectionParameters $clusterConnectionParameters)
+            return $certificate
```
Comment (bishal-pdMSFT) [inline Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1]: Where is this $certificate defined. Shouldn't this be returned from Add-Certificate first?

Comment (NCarlsonMSFT) [reply Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1]: Odd I thought I had added that, must have gotten lost in a local revision. Fixed now.

**Pull request merged**